src/cpp/session/modules/SessionProfiler.cpp

```diff
@@ -28,23 +36,52 @@ namespace modules {
 namespace profiler {
 
 namespace {
-      
 
+#define kProfilesCacheDir "profiles-cache"
+#define kProfilesUrlPath "profiles"
 
+std::string profilesCacheDir() 
+{
+   return module_context::scopedScratchPath().childPath(kProfilesCacheDir)
+      .absolutePath();
+}
+
+SEXP rs_profilesPath()
+{
+	r::sexp::Protect rProtect;
+	return r::sexp::create(profilesCacheDir(), &rProtect);
+}
 
-   
 } // anonymous namespace
-   
+
+void handleProfilerResReq(const http::Request& request,
+                            http::Response* pResponse)
+{
+   std::string resourceName = http::util::pathAfterPrefix(request, "/" kProfilesUrlPath "/");
+
+   core::FilePath profilesPath = core::FilePath(profilesCacheDir());
+   core::FilePath profileResource = profilesPath.childPath(resourceName);
+
+   pResponse->setCacheableFile(profileResource, request);
+}
+
 Error initialize()
 {  
    ExecBlock initBlock ;
+   
    initBlock.addFunctions()
-      (boost::bind(module_context::sourceModuleRFile, "SessionProfiler.R"));
+      (boost::bind(module_context::sourceModuleRFile, "SessionProfiler.R"))
+      (boost::bind(module_context::registerUriHandler, "/" kProfilesUrlPath "/", handleProfilerResReq));
+
+   // register rs_profilesPath
+   r::routines::registerCallMethod(
```

Comment:
You can use `RS_REGISTER_CALL_METHOD` for a shorthand way of registering routines: https://github.com/rstudio/rstudio/blob/master/src/cpp/r/include/r/RRoutines.hpp#L35-L43 (avoids a bit of the verbosity)